### PR TITLE
Chore: Addon documentation restructure

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the current addon documentation is restructured for clarity.

What was done:
- Extracted the addon documentation from the API section into it's own section (addons).
- Adjusted the TOC.
- Split the documentation into smaller and more logical pieces.
- Created the assets/snippets required to back the documentation.
- Updated the writing addons example.
- Updated links pointing at the old documentation.
- Updated Introduction.stories.mdx file to properly link to the right place in the documentation (lib/cli/src/frameworks/common/Introduction.mdx)

Example code for the writing addon [here](https://github.com/jonniebigodes/my-addon-example)

Feel free to provide feedback

### CODE REVIEW

File: docs/addons/addon-knowledge-base.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+{
+  "scripts":{
+    "start": "npm run build -- --watch",
+  }
+}
+```
+
+#### Local Storybook addons
+
+If you're developing a local Storybook addon (an addon built on top of a existing Storybook installation) HMR (hot module replacement) should be available out of the box. If you don't see the changes being reflected, add the flag `--no-manager-cache` to the `storybook` script and restart Storybook.
```
Comment (winkerVSbecks) [inline docs/addons/addon-knowledge-base.md:86]: 🙌

File: docs/addons/addon-types.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+title: 'Types of addons'
+---
+
+Each Storybook addon is classified into two general categories, UI-based or Presets. Documented here are some of the features available for each type of addon. Use it as a reference when creating your next Storybook addon.
+
+## UI-based addons
+
+This particular type of addons allows you to customize Storybook's UI. If the addon you're developing falls into this category you can use some of the following elements to further customize the interface.
+
+### Panels
```
Comment (winkerVSbecks) [inline docs/addons/addon-types.md:11]: Could you add images of these UI elements from @domyen's post? https://storybook.js.org/blog/addon-catalog-sneak-peak/

File: docs/addons/install-addons.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+With the exception of presets, all other Storybook addons follow the same set of installation instructions. For example, to include accessibility testing in Storybook, run the following command to install the necessary addon:
+
+```shell
+yarn add -D @storybook/addon-a11y
+```
+
+### Addon registration
+
+Installing the addon is not enough, it needs to be registered. Update [`.storybook/main.js`](../configure/overview.md#configure-story-rendering) to the following:
```
Comment (winkerVSbecks) [inline docs/addons/install-addons.md:17]: Minor language recommendation. Saying
> Installing the addon is not enough

makes the usage process sound more laborious. Could you reword this to something like:
> ### Using addons
> With the exception of presets, all other Storybook addons follow the same set of installation instructions. It's a 2 step process: install and register. 
>
> For example, to include accessibility testing in Storybook, run the following command to install the necessary addon:
>...

File: docs/addons/install-addons.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+For example, to use SCSS styling, first install the preset and the required dependencies:
+
+```sh
+yarn add -D @storybook/preset-scss css-loader sass-loader style-loader
+```
+
+### Preset registration
+
+Installing the preset and the required dependencies is not enough, as with the example above presets also need to be registered. Update [`.storybook/main.js`](../configure/overview.md#configure-story-rendering) to the following:
```
Comment (winkerVSbecks) [inline docs/addons/install-addons.md:50]: Same here. You could say that presets require a 3rd step

File: docs/addons/install-addons.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+<div class="aside">
+Other addons like the one mentioned might require additional configurations. Check the addon README for additional configuration requirements.
+</div>
+
+That's it, when Storybook starts up accessibility testing is enabled.
+
+![Storybook addon installed and registered](./storybook-addon-installed-registered.png)
+
+
+### Install Storybook presets
```
Comment (winkerVSbecks) [inline docs/addons/install-addons.md:38]: Let's use consistent terminology and call them `Preset addons`

File: docs/addons/introduction.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+#### UI-based addons
+
+This particular type of addons focuses mainly on the interface. Installing addons that fall into this category help you with your existing workflows. Good examples of this particular type are: [Controls](../essentials/controls.md), [Docs](../writing-docs/introduction.md) or the [Accessibility addon](https://github.com/storybookjs/storybook/tree/master/addons/a11y).
+
+<div class="aside">
+Read our <a href="./writing-addons">documentation</a> to learn more about creating addons.
+</div>
+
+#### Presets
```
Comment (winkerVSbecks) [inline docs/addons/introduction.md:39]: `Preset addons`

File: docs/addons/writing-addons.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
+---
+title: 'Writing your own Storybook addon'
+---
+
+One of Storybook's main features it's is robust addon ecosystem. Addons can be used to enhance your developer experience or workflows. We're going to guide you with the necessary steps to create your own first addon.
```
Comment (winkerVSbecks) [inline docs/addons/writing-addons.md:5]: `is it's`

File: docs/addons/writing-addons.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+
+<CodeSnippets
+  paths={[
+    'common/storybook-addon-panel-initial.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+This is boilerplate code for any addon that adds a panel to Storybook, and there’s really not much going on here. In this case, we’re just adding a static `div` that renders when the panel is selected in Storybook’s UI.
```
Comment (winkerVSbecks) [inline docs/addons/writing-addons.md:117]: I'm going to argue there is quite a bit happening here. We could say it's boilerplate. But, let's explain what exactly is happening. i.e., `register`, `add` and the `key` bit which is super important and without it you get errors.

File: docs/addons/writing-addons.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+
+- A `package.json file` with metadata about the addon.
+- Peer dependencies of `react` and `@storybook/addons`.
+- A `register.js` file at the root level written as an ES5 module.
+- A `src` directory containing the ES6 addon code.
+- A `dist` directory containing transpiled ES5 code on publish.
+- A [GitHub](https://github.com/) account to host your code.
+- A [NPM](https://www.npmjs.com/) account to publish the addon.
+
+For a good template of an addon that meets the requirements above, check out [@storybook/addon-controls](https://npmjs.com/package/@storybook/addon-controls).
```
Comment (winkerVSbecks) [inline docs/addons/writing-addons.md:215]: I think we should find a standalone example. `@storybook/addon-controls` lives in the monorepo and relies on the monorepo's build system. It's really hard to figure out how the addon is actually built and published.

**Review by winkerVSbecks — CHANGES_REQUESTED** [review]
@jonniebigodes we should add a reference to `@storybook/components` on the API page

File: docs/addons/writing-addons.md (hunk 9 above)
Comment (jonniebigodes) [reply docs/addons/writing-addons.md:215]: @winkerVSbecks we could use the one you've built? WDYT? It's a standalone one outside the monorepo.

File: docs/addons/addon-types.md (hunk 2 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md:11]: @winkerVSbecks probably better to use an image that matches the code snippet for accuracy. I'm going to commit them shortly.

File: docs/addons/introduction.md (hunk 6 above)
Comment (jonniebigodes) [reply docs/addons/introduction.md]: Updated with the latest commit

File: docs/addons/install-addons.md (hunk 5 above)
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Updated with the latest commit

File: docs/addons/writing-addons.md (hunk 8 above)
Comment (jonniebigodes) [reply docs/addons/writing-addons.md]: Updated with the latest commit.

File: docs/addons/writing-addons.md (hunk 7 above)
Comment (jonniebigodes) [reply docs/addons/writing-addons.md]: Thanks for catching that one. Updated with the latest commit

File: docs/addons/install-addons.md (hunk 4 above)
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Updated with the latest commit.

File: docs/addons/install-addons.md (hunk 3 above)
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Updated with the latest commit

File: docs/addons/addon-knowledge-base.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md:86]: This small tweak can save a lot of time :)

File: docs/addons/addon-types.md (hunk 2 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md:11]: Images are added with the latest commits.

**Comment by jonniebigodes** [comment]
With the latest commits the documentation and toc are updated to address the initial feedback.

File: docs/addons/install-addons.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
+---
+title: Set up Storybook addons
+---
+
+Storybook boasts a wide variety of addons in it's ecosystem. All of them packaged as NPM modules and listed in our [addon gallery](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
```
Comment (shilman) [inline docs/addons/install-addons.md:5]:
```suggestion
Storybook boasts a wide variety of addons in its ecosystem. All of them packaged as NPM modules and listed in our [addon gallery](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
```

**Review by shilman — CHANGES_REQUESTED** [review]
I think we should make every addon a preset addon. It creates an extra step for addon authors, but it simplifies things for addon users. If you look at the essential addons, they all use presets at this point to add default parameters or decorators (and more). I believe most user addons can also make use of this. WDYT? cc @domyen @winkerVSbecks We can do this in a separate PR for expediency.

**Comment by domyen** [comment]
@shilman re:preset addons, I think that's a good idea. 
Can we identify that in our backend and rank preset addons higher (to encourage their use)? Is the recommendation to call out the preset pattern in the documentation more clearly?

File: docs/addons/install-addons.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
+---
+title: Set up Storybook addons
+---
+
+Storybook boasts a wide variety of addons in its ecosystem. All of them packaged as NPM modules and listed in our [addon gallery](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
```
Comment (domyen) [inline docs/addons/install-addons.md:5]:
```suggestion
Storybook boasts a wide variety of addons in its ecosystem. All of them packaged as NPM modules and listed in our [addon catalog](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
```

File: docs/addons/introduction.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
+---
+title: 'Overview of Storybook addons'
+---
+
+Addons extend Storybook with features and integrations that are not built into the core. Most Storybook features are implemented as addons. For instance: [documentation](../writing-docs/introduction.md), [accessibility testing](https://github.com/storybookjs/storybook/tree/master/addons/a11y), [interactive controls](../essentials/controls.md), among others.
+The [addon API](./addons-api.md) makes it easy for you to configure and customize Storybook in new ways. There are countless addons made by the community that unlock time-saving workflows.
+
+Browse our [addon gallery](/addons) to install an existing addon or as inspiration for your own addon.
```
Comment (domyen) [inline docs/addons/introduction.md:8]:
```suggestion
Browse our [addon catalog](/addons) to install an existing addon or as inspiration for your own addon.
```

File: docs/addons/addon-types.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
+---
+title: 'Types of addons'
+---
+
+Each Storybook addon is classified into two general categories, UI-based or Presets. Documented here are some of the features available for each type of addon. Use it as a reference when creating your next Storybook addon.
```
Comment (domyen) [inline docs/addons/addon-types.md:5]:
```suggestion
Each Storybook addon is classified into two general categories, UI-based or Presets. Each type of addons features are documented here. Use this as a reference when creating your addon.
```

File: docs/addons/addon-types.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+
+## UI-based addons
+
+This particular type of addons allows you to customize Storybook's UI. If the addon you're developing falls into this category you can use some of the following elements to further customize the interface.
```
Comment (domyen) [inline docs/addons/addon-types.md:9]:
```suggestion
UI-based addons allow you to customize Storybook's UI using the following elements.
```

Edit for conciseness. This can be shortened into one sentence. 

Also be judicious about "some", that word is imprecise and doesn't give the reader confidence in the docs.

File: docs/addons/addon-types.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+### Panels
+
+This type of UI element allows you to add your own custom <code>Panel</code> to Storybook's UI. Currently the most common use case available throughout our ecosystem. Good examples that use this type of interface are: [@storybook/actions](../essentials/actions.md) and [@storybook/a11y](https://github.com/storybookjs/storybook/tree/next/addons/a11y).
```
Comment (domyen) [inline docs/addons/addon-types.md:13]:
```suggestion
Panel addons allow you to add your own UI in Storybook's addon panel. This is the most common type of addon in the ecosystem. For example, the official [@storybook/actions](../essentials/actions.md) and [@storybook/a11y](https://github.com/storybookjs/storybook/tree/next/addons/a11y) use this pattern.
```

File: docs/addons/addon-types.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+![Storybook panel](./storybook-panel.png)
+
+Below is the boilerplate code that you can use in your own addon to add a `Panel` in Storybook's UI.
```
Comment (domyen) [inline docs/addons/addon-types.md:17]: Rephrase this doc to describe what the reader should do in fewer words. Get straight to the point.

For example:
"Below is the boilerplate code that you can use in your own addon to add a `Panel` in Storybook's UI."

"Add a `Panel` to Storybook's UI using this boilerplate code."

**Review by domyen — CHANGES_REQUESTED** [review]

**Comment by shilman** [comment]
@domyen good call on ranking those addons higher 👍  we can probably detect that

Right now the documentation talks about making a `/register` entry and registering your addon in `main.js` using that.  Even tho that's sufficient for some addons, I could imagine restructuring the documentation that says something more along the lines of:

> Now let’s add a panel to Storybook. Create a new directory called `src` and inside a new file called `preset.js`. Presets are the way your addon hooks into Storybook: in the preset we can can add to the Storybook UI, add to the user's preview iframe, modify babel and webpack settings, and more.  For now we'll just modify the Storybook UI: ...

Even though this is adds an extra step to the process (you still need to create the same `register.js` file after you create the preset) it reinforces the idea of presets and ensures that new addons will use them. It will also be easier to educate addon authors to automate more with the preset if they already have a trivial one that's part of their project.

**Comment by domyen** [comment]
Thanks @jonniebigodes! 

I left some representative feedback on the tone and style of the new docs. It's only in a few places right now, can you apply that feedback to the rest of the doc.

The gist is to make the writing more precise and use fewer words. Folks tend to skim the docs – so having a high ratio of ideas to words makes reading easier.

**Review by winkerVSbecks — APPROVED** [review]

**Comment by jonniebigodes** [comment]
So that we're on the same page here and i understand this fully. Regarding the writing addon documentation here's what is happening:
1. Project is initialized
2. Packages (react and storybook) installed
3. Babel is configured
4. Build script
5. Entry point for the addon (`src/register.js`) is created.
6. Addon is updated to include the parameter supplied with any given story.
7. The root level `register.js` is created to reference compiled addon.

@shilman based on the comment above then a new step would be included after item 5, correct? Asides from the `register.js` the reader would also create a `src/preset.js` with the following inside:

```js
// my-addon/src/preset.js
export function managerEntries(entry = []) {
  return [...entry, require.resolve('./register')]; 
}
```

Meaning that the item 7 would be changed, we would use a root level `preset.js` instead of the root level `register.js`. Which in a nutshell we're turning the addon into more of a preset.

Let me know so that i can adjust accordingly.

**Comment by shilman** [comment]
Yeah that sounds good @jonniebigodes . A slight variation would be to include preset before step 5. I think either would work.

**Comment by jonniebigodes** [comment]
Ok, thanks for the clarification, i'm going to adjust the page.

File: docs/addons/addon-types.md (hunk 13 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md]: Fixed with the latest commit

File: docs/addons/addon-types.md (hunk 15 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md]: Fixed with the latest commit

File: docs/addons/addon-types.md (hunk 16 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md]: Fixed with the latest commit

File: docs/snippets/common/my-addon-root-level-preset.js.mdx [hunk 17]
```diff
@@ -1,5 +1,5 @@
 ```js
-// /myaddon/register.js
+// /my-addon/preset.js
 
 module.exports = require('./dist/register.js');
```
Comment (shilman) [inline docs/snippets/common/my-addon-root-level-preset.js.mdx:4]:
```suggestion
module.exports = require('./dist/preset.js');
```

**Review by shilman — CHANGES_REQUESTED** [review]
Typo

File: docs/snippets/common/my-addon-root-level-preset.js.mdx (hunk 17 above)
Comment (jonniebigodes) [reply docs/snippets/common/my-addon-root-level-preset.js.mdx]: fixed with the latest commit.

File: docs/snippets/common/my-addon-storybook-registration.js.mdx [hunk 18]
```diff
@@ -0,0 +1,13 @@
+```js
+// /my-addon/.storybook/main.js
+
+module.exports = {
+  stories: [],
+  addons: [
+    "@storybook/addon-links",
+    "@storybook/addon-essentials",
+    "@storybook/preset-create-react-app",
+    "../src/register.js" //👈 Our addon registered here
```
Comment (shilman) [inline docs/snippets/common/my-addon-storybook-registration.js.mdx:10]:
```suggestion
    "../src/preset.js" //👈 Our addon registered here
```

**Review by shilman — CHANGES_REQUESTED** [review]
Another one. I've also updated the outline addon to reflect this idea and its  directory structure is a bit different. WDYT? cc @winkerVSbecks 

https://github.com/chromaui/storybook-outline/pull/7

File: docs/addons/writing-addons.md [hunk 19]
```diff
@@ -0,0 +1,250 @@
...
+    'common/my-addon-preset-implementation.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Presets are the way addons hook into Storybook. Among other tasks they allow you to:
+
+- Add to [Storybook's UI](#add-a-panel).
+- Add to the [preview iframe](./writing-presets.md#preview-entries).
```
Comment (domyen) [inline docs/addons/writing-addons.md:115]:
```suggestion
- Add to the [preview iframe](./writing-presets.md#preview-entries)
```

File: docs/addons/writing-addons.md [hunk 20]
```diff
@@ -0,0 +1,250 @@
...
+- Modify [babel](./writing-presets.md#babel) and [webpack settings](./writing-presets.md#webpack).
```
Comment (domyen) [inline docs/addons/writing-addons.md:116]:
```suggestion
- Modify [babel](./writing-presets.md#babel) and [webpack settings](./writing-presets.md#webpack)
```

File: docs/addons/writing-addons.md [hunk 21]
```diff
@@ -0,0 +1,250 @@
...
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+Make sure to include the <code>key</code> when you register the addon. This will prevent any issues when the addon renders.
+</div> 
+
+Going over the code snippet in more detail. When Storybook starts up:
+
+- It [registers](./addons-api.md#addonsregister) the addon.
```
Comment (domyen) [inline docs/addons/writing-addons.md:140]: Make this a numbered list and remove the periods

File: docs/addons/writing-addons.md [hunk 22]
```diff
@@ -0,0 +1,250 @@
...
+
+<CodeSnippets
+  paths={[
+    'common/my-addon-root-level-preset.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Making this change helps Storybook's addon registration. By design Storybook looks for either a `preset.js` or a `register.js` file located at the root level. Omitting this file would require additional configuration from the addon consumer.
```
Comment (domyen) [inline docs/addons/writing-addons.md:225]:
```suggestion
This auto-registers the addon without any additional configuration from the user. Storybook looks for either a `preset.js` or a `register.js` file located at the root level.
```

File: docs/addons/writing-addons.md [hunk 23]
```diff
@@ -0,0 +1,250 @@
...
+
+### Packaging and publishing
+
+With the example above we showed how to create a bare-bones addon. Before publishing and distributing your own addon make sure you have met the following requirements:
```
Comment (domyen) [inline docs/addons/writing-addons.md:229]:
```suggestion
Now that you've seen how to create a bare-bones addon, let's see how to share it with the community. Before we begin, make sure your addon meets the following requirements:
```

File: docs/addons/writing-addons.md [hunk 24]
```diff
@@ -0,0 +1,250 @@
...
+
+- A `package.json file` with metadata about the addon.
```
Comment (domyen) [inline docs/addons/writing-addons.md:231]: - Remove periods (these aren't sentences)
- No need to repeat "A..." for every item

File: docs/addons/writing-addons.md [hunk 25]
```diff
@@ -0,0 +1,250 @@
...
+- Peer dependencies of `react` and `@storybook/addons`.
+- A `preset.js` file at the root level written as an ES5 module.
+- A `src` directory containing the ES6 addon code.
+- A `dist` directory containing transpiled ES5 code on publish.
+- A [GitHub](https://github.com/) account to host your code.
+- A [NPM](https://www.npmjs.com/) account to publish the addon.
+
+For a good template of an addon that meets the requirements above, check out [storybook-addon-outline](https://www.npmjs.com/package/storybook-addon-outline).
```
Comment (domyen) [inline docs/addons/writing-addons.md:239]:
```suggestion
For example, check out [storybook-addon-outline](https://www.npmjs.com/package/storybook-addon-outline) to see a project that meets these requirements.
```

File: docs/addons/writing-addons.md [hunk 26]
```diff
@@ -0,0 +1,250 @@
...
+
+### Next steps
```
Comment (domyen) [inline docs/addons/writing-addons.md:241]:
```suggestion
### More guides and tutorials
```

File: docs/addons/writing-addons.md [hunk 27]
```diff
@@ -0,0 +1,250 @@
...
+
+In the previous example, we introduced the structure of an addon, but barely scratched the surface of what addons can do.
+
+To dive deeper we recommend [Learn Storybook’s “creating addons”](https://www.learnstorybook.com/intro-to-storybook/react/en/creating-addons/) tutorial. It’s an excellent walkthrough that covers the same ground as the above introduction, but goes further and leads you through the full process of creating a realistic addon. Or [how to build a Storybook addon](https://www.chromatic.com/blog/how-to-build-a-storybook-addon/) blog post that guides you through the steps of creating a standalone addon in great length.
```
Comment (domyen) [inline docs/addons/writing-addons.md:245]:
```suggestion
To dive deeper we recommend [Learn Storybook’s “creating addons”](https://www.learnstorybook.com/intro-to-storybook/react/en/creating-addons/) tutorial. It’s an excellent walkthrough that covers the same ground as the above introduction, but goes further and leads you through the full process of creating a realistic addon. 

[How to build a Storybook addon](https://www.chromatic.com/blog/how-to-build-a-storybook-addon/) shows you how to create a standalone addon in great detail.
```

File: docs/addons/writing-presets.md [hunk 28]
```diff
@@ -1,8 +1,8 @@
 ---
-title: 'Writing your own Storybook Preset'
+title: 'Writing your own Storybook preset addon'
```
Comment (domyen) [inline docs/addons/writing-presets.md:2]:
```suggestion
title: 'Write a Storybook preset addon'
```

**Review by domyen — CHANGES_REQUESTED** [review]
Looking good! Added some rfcs for grammar, punctuation, and clarity.

File: docs/addons/addon-knowledge-base.md [hunk 29]
```diff
@@ -0,0 +1,119 @@
+---
+title: Addons knowledge base
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 30]
```diff
@@ -0,0 +1,119 @@
...
+---
+
+Once you understand the basics of writing addons, there are a variety of common enhancements to make your addon better. This page details additional information about addon creation. Use it as a quick reference guide when creating your own addons. 
+
+### Disabling the addon panel
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 31]
```diff
@@ -0,0 +1,119 @@
...
+
+<CodeSnippets
+  paths={[
+    'react/button-story-disable-addon.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+### Styling your addon
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 32]
```diff
@@ -0,0 +1,119 @@
...
+| Color Pallette (see note below) | [See implementation](https://github.com/storybookjs/storybook/tree/master/lib/components/src/Colors)         | [See story](https://5a375b97f4b14f0020b0cda3-wbeulgbetj.chromatic.com/?path=/story/basics-colorpalette--page) |
+| Icon                            | [See implementation](https://github.com/storybookjs/storybook/blob/master/lib/components/src/icon/icons.tsx) | [See story](https://5a375b97f4b14f0020b0cda3-wbeulgbetj.chromatic.com/?path=/story/basics-icon--labels)       |
+| Typography                      | [See implementation](https://github.com/storybookjs/storybook/tree/master/lib/components/src/typography)      | [See story](https://5a375b97f4b14f0020b0cda3-wbeulgbetj.chromatic.com/?path=/story/basics-typography--all)    |
+
+
+<div class="aside">
+The color pallette implemented by <code>@storybook/components</code> is a high level abstraction of the implementation of <a href="https://github.com/storybookjs/storybook/tree/next/lib/theming/src"><code>@storybook/theming</code></a> package.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 33]
```diff
@@ -0,0 +1,119 @@
...
+</div>
+
+### Build system
+
+When you are developing your addon as a package, you can’t use `npm link` to add it to your project. List  your addon as a local dependency into your package.json:
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 34]
```diff
@@ -0,0 +1,119 @@
...
+    "@storybook/addon-controls": "file:///home/username/myrepo"
+  }
+}
+```
+
+<div class="aside">
+Run either <code>yarn</code> or <code>npm install</code> to install the addon.
+</div>
+
+### Hot module reload
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 35]
```diff
@@ -0,0 +1,119 @@
...
+
+While developing your addon you can configure HMR (hot module replacement ) to reflect the changes made.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 36]
```diff
@@ -0,0 +1,119 @@
...
+{
+  "scripts":{
+    "start": "npm run build -- --watch",
+  }
+}
+```
+
+#### Local Storybook addons
+
+If you're developing a local Storybook addon (an addon built on top of a existing Storybook installation) HMR (hot module replacement) should be available out of the box. If you don't see the changes being reflected, add the flag `--no-manager-cache` to the `storybook` script and restart Storybook.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md]: Fixed with the latest commit.

File: docs/addons/addon-types.md [hunk 37]
```diff
@@ -0,0 +1,90 @@
...
+  paths={[
+    'common/storybook-addon-tab-example.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+
+<div class="aside">
+If want to learn more about writing your own addon and include some of these elements, read our <a href="./writing-addons">documentation</a>.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-types.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 38]
```diff
@@ -0,0 +1,89 @@
+---
+title: Set up Storybook addons
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 39]
```diff
@@ -0,0 +1,89 @@
...
+---
+
+Storybook boasts a wide variety of addons in its ecosystem. All of them packaged as NPM modules and listed in our [addon catalog](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
+
+### Using addons
+
+With the exception of preset addons, all other follow the same set of installation instructions. Based on a two step process: install and register.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 40]
```diff
@@ -0,0 +1,89 @@
+---
+title: Set up Storybook addons
+---
+
+Storybook boasts a wide variety of addons in its ecosystem. All of them packaged as NPM modules and listed in our [addon catalog](/addons). We're going to guide you through the necessary steps to install and register Storybook addons.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 41]
```diff
@@ -0,0 +1,89 @@
...
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+Other addons like the one mentioned might require additional configurations. Check the addon README for additional configuration requirements.
+</div>
+
+That's it, when Storybook starts up accessibility testing is enabled.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 42]
```diff
@@ -0,0 +1,89 @@
...
+
+![Storybook addon installed and registered](./storybook-addon-installed-registered.png)
+
+
+### Using preset addons 
+
+Storybook preset addons are grouped collections of specific `babel`,`webpack` and `addons` configurations for distinct use cases. Each one with it's own set of instructions. Based on a three step process: install, register and optionally configuration.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/install-addons.md [hunk 43]
```diff
@@ -0,0 +1,89 @@
...
+
+That's it, when Storybook starts up, it will configure itself to use SCSS styling without any further configuration.
+
+
+
+#### Optional configuration
+
+Most preset addons can also take additional parameters. The most common use cases are:
+
+- Addon configuration.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/install-addons.md]: Fixed with the latest commit.

File: docs/addons/addon-knowledge-base.md [hunk 44]
```diff
@@ -0,0 +1,119 @@
...
+Run either <code>yarn</code> or <code>npm install</code> to install the addon.
+</div>
+
+### Hot module reload
+
+While developing your addon you can configure HMR (hot module replacement ) to reflect the changes made. 
+
+#### Standalone Storybook addons
+
+If you're developing a standalone addon, add a new script to `package.json` with the following:
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/addons/addon-knowledge-base.md:107]: Addons that are installed by the reader/user through their package manager of choice and not implemented inside the local `.storybook` folder

**Review by shilman — APPROVED** [review]
💯
